proto/spire/api/server/trustdomain/v1/trustdomain.proto

```diff
@@ -73,9 +77,13 @@ message GetFederationRelationshipRequest {
     // (e.g., "example.org").
     string trust_domain = 1;
 
+    // If true, it returns the static federation relationship defined in the
+    // SPIRE server configuration file.
+    bool include_static = 2;
+
```

Comment:
Hmm, I wonder about this behavior. If you call `Get` with `include_static=true` and no static entry exists, do you get back NOT_FOUND or the dynamic entry (assuming there is a dynamic entry)?

Sorry to rehash this all again, but I wonder.... should we just simply have separate RPCs for the static relationships? e.g., `GetStaticFederationRelationship` and `ListStaticFederationRelationships` (there would be no create/update/delete counterparts)

Reply:
I thought about the NOT_FOUND case when considering the introduction of this field. I think that there are some pros / cons with each approach.
One thing that I don't love about having separate RPCs is that in that case you have to make multiple calls if you want to list the full set of relationships and join the results.
With a single RPC and a field you can get all the relationships in a single call but as you pointed the semantics is less clear in the NOT_FOUND case.

I don't really have a strong opinion. Just wanted to point one disadvantage of going with multiple RPCs. It also may not be super obvious that `ListFederationRelationships` will return dynamic entries only, by looking at its name. So we may need to rename the RPCs to reflect that they will deal with dynamic entries only.

Reply:
> If you call `Get` with `include_static=true` and no static entry exists, do you get back NOT_FOUND or the dynamic entry (assuming there is a dynamic entry)?

I would expect to get NOT_FOUND in that case.

I am ok with both approaches. If the concern is only about clarity, another option could be to use a `data_source` attribute for Get and List that could be set to `static`, `dynamic` and `static_and_dynamic` using an enum.

GetFederationRelationships would fail if invoked with `static_and_dynamic` (or one can take precedence over the other)